src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
+const spiderElement = document.querySelector('.spider');
+const wallBorder = document.querySelector('.wall');
+
 document.addEventListener('click', (e) => {
   // write code here
+  const spiderWidth = spiderElement.offsetWidth;
+  const spiderHeight = spiderElement.offsetHeight;
+
+  const coordOfWall = wallBorder.getBoundingClientRect();
+
+  let xPosition = e.clientX - coordOfWall.left - spiderWidth / 2;
+  let yPosition = e.clientY - coordOfWall.top - spiderHeight / 2;
```

Comment:
Ensure that the calculation of `xPosition` and `yPosition` accounts for any potential changes in the wall's position or dimensions. If the wall's position or size can change dynamically, consider recalculating `coordOfWall` inside the event listener to ensure accurate positioning.